data/helpers.d/setting

```diff
@@ -158,12 +158,22 @@ ynh_webpath_register () {
 
 # Create a new permission for the app
 #
-# example: ynh_permission_create --permission=admin --url=/admin --allowed="alice bob"
-#
-# usage: ynh_permission_create --permission "permission" [--url=url] [--allowed="group1 group2"]
-# | arg: -p, --permission=  - the name for the permission (by default a permission named "main" already exist)
-# | arg: -u, --url=         - (optional) URL for which access will be allowed/forbidden
-# | arg: -a, --allowed=     - (optional) A list of group/user to allow for the permission
+# example: ynh_permission_create --permission=admin --url=/admin --additional_urls=domain.tld/otherurl /superadmin --allowed=alice bob --label="My app admin"
+#
+# usage: ynh_permission_create --permission="permission" [--url="url"] [--additional_urls="second-url" [ "third-url" ]] [--auth_header=true|false]
+#                                                        [--allowed=group1 [ group2 ]] [--label="label"] [--show_tile=true|false]
+#                                                        [--protected=true|false]
+# | arg: -p, permission=        - the name for the permission (by default a permission named "main" already exist)
+# | arg: -u, url=               - (optional) URL for which access will be allowed/forbidden
+# | arg: -A, additional_urls=   - (optional) List of additional URL for which access will be allowed/forbidden
+# | arg: -h, auth_header=       - (optional) Define for the URL of this permission, if SSOwat pass the authentication header to the application. Default is true
+# | arg: -a, allowed=           - (optional) A list of group/user to allow for the permission
+# | arg: -l, label=             - (optional) Define a name for the permission. This label will be shown on the SSO and in the admin.
+# |                                     Default is "APP_LABEL (permission name)".
+# | arg: -t, show_tile=         - (optional) Define if a tile will be shown in the SSO
+# | arg: -P, protected=         - (optional) Define if this permission is protected. If it is protected the administrator
+# |                             won't be able to add or remove the visitors group of this permission.
+# |                             By default it's 'true' (for the permission different than 'main').
```

Comment:
> By default it's 'true'

Does that mean any permission added by an app will be unmovable by an admin by default ?

Reply:
Yes

Reply:
It may sound completely silly for sure, but when taking some decision that are actually completely app related, it sounds a little bit legit to talk with the apps group...

And actually, it does not make any sense to forbid users to modify the permission by default for all apps.
For sure we can have some specific situations, and probably that option may be handy.
But to not forbid by default make just no sense !

Reply:
Well,

Note that it's protect only to add/remove the visitor on this permission. Also the main permission is not protected by default.

I forbid by default because I thought that probably In some case the packager will create a permission to unprotected by example the API. For the permission without uris it don't make sens to not protect it as playing with the visitor group on a permission without uris has no effect.

But yes we should discuss.

Reply:
> probably In some case

Which sound clearly up to the packager to decide whether or not !

Unless I missed something this is not about a permission without uris since a few lines before there's `-u, url=`
If you do consider that a permission without uris (does not make any sense for me right now...) or an API should be protected, then add a comment explaining that.
To be honest, the comments in the header of that helper, which looks like a very important helper, are really not helpful to actually understand what to do with the different options it has.
- `-u, url=` should give clear examples of what you can do with it.
- `additional_urls=` makes no sense for me. Why there's a second argument ?
- Considering `url=` is optional !? What other usage you could have ? This is about the web access !
- `auth_header=` Yet I know how the SSO work, this is still not completely clear for me. And I know the mechanism... What to expect for other ?
I guess this is actually skipped_uris. Which means it didn't work... But anyway... -_-
- `label=` and `show_tile=` Those 2 are partially linked and that should be explicitly explained with concrete example.
- `protected=` Somehow clear on what it does but not why you would use it. Example again should be given.

But `protected`, I have no doubt we can have use cases, even if I have none in mind. But that's clearly does not sound like the default behavior a packager would want for a permission other than main.
Currently most use case we have for extra permissions are generated links for downloading files or images. I see no reason for an admin to not want to have them private for a restricted circle of users.
And API indeed, which again could totally be turn private by an user not using them and seeing (which is quite right) a potential security breach.
And yet, such users may change their mind and turn it public again afterward.

Reply:
> But `protected`, I have no doubt we can have use cases, even if I have none in mind. But that's clearly does not sound like the default behavior a packager would want for a permission other than main.
> Currently most use case we have for extra permissions are generated links for downloading files or images. I see no reason for an admin to not want to have them private for a restricted circle of users.
> And API indeed, which again could totally be turn private by an user not using them and seeing (which is quite right) a potential security breach.
> And yet, such users may change their mind and turn it public again afterward.

Why I added this:
https://github.com/YunoHost/issues/issues/1486
https://forum.yunohost.org/t/yunohost-3-7-spooky-testing-call-for-feedback/9385/30

Reply:
Huh yes...
Well I read
> It would be nice to have a system to forbid adding a permission to "visitors"

I do agree, it's nice to have that possibility, but I'm not saying that's a bad idea to protect a permission, I'm not saying that argument shouldn't exist.
I'm saying that it should not be forbidden by default.
"To be nice to" doesn't mean, let's impose that to everyone !

Reply:
Ok, so I'll disable it by default.